hnne/projector.py

```diff
@@ -86,13 +94,32 @@ class HNNE(BaseEstimator):
     """
     def __init__(
             self,
-            dim: int = 2,
+            dim: Union[Optional[int], None] = None,
```

Comment:
Here I would keep the type as int and set its default to 2. It is nice to have this as a default as it is the most common case.